### PR TITLE
Implement reminder if using run/run-script without 'install'

### PR DESCRIPTION
**Pull request by timkrins**
For #10032

Implements warning if using run/run-script with locally defined dependencies but without a node modules directory.

Need help writing tests - couldn't figure out how to achieve a pass/fail through using npm in tap.

### CODE REVIEW

File: lib/run-script.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ module.exports = runScript
 var lifecycle = require('./utils/lifecycle.js')
 var npm = require('./npm.js')
 var path = require('path')
+var fs = require('fs')
```
Comment (KenanY) [inline lib/run-script.js]: Might as well use `graceful-fs` I think.

**Comment by KenanY** [comment]
I wonder if this is best done [here](https://github.com/npm/npm/blob/8946cb3f321995792a3009f6372a6f87e81873ed/lib/utils/lifecycle.js#L240-L251) so that this warning only prints if a script actually failed to run.

Also would:

``` js
fs.stat(path.resolve(pkgdir, 'node_modules/'), ...)
```

be "better" than having to do `fs.readdir` then the `indexOf` check?

**Comment by othiym23** [comment]
@KenanY Agreed. @timkrins if you can make that change, this will get merged shortly thereafter. Thanks!

**Comment by timkrins** [comment]
Will get onto it tomorrow, cheers!

**Comment by iarna** [comment]
This landed in 3.6.0! =)

**Comment by thomblake** [comment]
@timkrins I just encountered this, great work, very helpful!

**Pull request closed without merging**